File: 3164-last-visited-integers/3164-last-visited-integers.cpp
class Solution {
public:
    vector<int> lastVisitedIntegers(vector<string>& words) {
        int n = words.size();
    
        vector<int> ans;
        vector<int> nums;  
        int p_Count = 0; 

        for (int i = 0; i<n; i++) {
            string word = words[i];
            if (word == "prev") {
                p_Count++;
                if (p_Count > nums.size()) {
                    ans.push_back(-1);
                } else {
                    ans.push_back(nums[nums.size() - p_Count]);
                }
            } else {
                int num = stoi(word);
                nums.push_back(num);
                p_Count = 0;
            }
        }

        return ans;
    }
};

File: 3164-last-visited-integers/3164-last-visited-integers.java
class Solution {
    public List<Integer> lastVisitedIntegers(List<String> words) {
        final List<Integer> answers = new ArrayList<>();
        final List<Integer> numbers = new ArrayList<>();
        
        int prevSeen = 0;
        
        for(String word: words){
            if("prev".equals(word)){
                prevSeen++;
                
                int i = numbers.size() - prevSeen;
                
                answers.add(0 <= i ? numbers.get(i) : -1);
            } else {
                prevSeen = 0;
                numbers.add(Integer.valueOf(word));
            }
        }
        
        return answers;
    }
}